uwblueprint/focus-on-nature
# Personal info registration

## Notion ticket link

[Registration Form: Personal Info Step view only components & Add/Remove camper ](https://www.notion.so/uwblueprintexecs/Registration-Form-Personal-Info-Step-view-only-components-Add-Remove-camper-6645744f9ff248378376d8da67fbe787)



## Implementation description
* Implemented personal info registration state by using `setCampers` useState. I directly edit the `campers` object
* `camperReducer` handles getting callbacks from `PersonalInfo/index.tsx` and handles updating `campers` object
* `checkPersonalInfoFilled` is a helper function for checking that the personal info page is filled correctly
* `usePersonalInfoHook` is a wrapper function to make calling `setCampers` easier. It provides a dispatch function which `camperCard` and `contactCard` use to update state.



## Steps to test
1. Check for camper add, update, delete
3.  Check for contact add, update, delete
4. Check that the `campers` object will be updated properly
5. Check resizing between PC, tablet, and phone



## What should reviewers focus on?
* The implementation has very tight coupling with the `campers` objects because we directly use it as our source of state. I assume that this is fine because we're going to ultimately export the `campers` object after registration is done
* Along the way, I took a short cut to make development faster. The shortcut was copying over the contacts of the 1st camper to all newly added campers. 
* The padding and resizing using chakraUI is pretty involved in this PR. I'm not fully sure if I could've removed any duplicate code or used more concise syntax.

[React App (1).webm](https://user-images.githubusercontent.com/117593511/206918069-3d09ebcc-4447-4d73-8412-e31a2f00f028.webm)

## Checklist
